api/post.go

```diff
@@ -55,11 +56,15 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
 
 		return
 	} else {
+		if result := <-Srv.Store.Channel().UpdateLastViewedAt(post.ChannelId, c.Session.UserId); result.Err != nil {
```

Comment:
No they won't mark channels as unread. Joining and leaving channels just calls CreatePost directly and bypasses createPost so it won't hit this function at all.